postgraas_server/postgres_instance_driver.py

```diff
@@ -55,29 +50,51 @@ def create_postgres_instance(postgraas_instance_name, connection_dict):
     if check_container_exists(postgraas_instance_name):
         raise ValueError('Container exists already')
     image = 'postgres:9.4'
-    port_bindings = {internal_port: connection_dict['port']}
-    restart_policy = {"Name": "unless-stopped"}
-    host_config_dict = c.create_host_config(port_bindings=port_bindings, restart_policy=restart_policy)
-    container_info = c.create_container(image,
-                                        name=postgraas_instance_name,
-                                        ports=[internal_port],
-                                        environment=environment,
-                                        labels={"postgraas": image},
-                                        host_config=host_config_dict)
-    container_id = container_info['Id']
-    c.start(container_id)
-    return container_id
+    container = c.containers.create(image,
+                                    name=postgraas_instance_name,
+                                    ports={internal_port: connection_dict['port']},
+                                    environment=environment,
+                                    restart_policy={"Name": "unless-stopped"},
+                                    labels={"postgraas": image})
+    container.start()
+    return container.id
 
 
 def delete_postgres_instance(container_id):
-    c = docker.Client(base_url='unix://var/run/docker.sock',
-                        timeout=30,
-                        version='auto')
-    c.remove_container(container_id, force=True)
-    return
+    c = _docker_client()
+    c.containers.get(container_id).remove(force=True)
+
+
+def wait_for_postgres_listening(container_id, timeout=10):
+    """
+    Inspect/follow the output of the docker container logs until encountering the
+    message from postgres that it accepts connections.
+
+    Raises in case the container does not exist.
+
+    Caveat: this check is specific to the used docker image.
+    Tested for current postgres:9.4 image.
+
+    :returns: boolean, whether the message has been (True), or the timeout has been hit (False).
+    """
+    c = _docker_client()
+    cont = c.containers.get(container_id)
+    for i in range(max(int(timeout * 10), 1)):
+        # not very efficient to always gather all logs, but this method
+        # is currently used for testing only, so it should be OK.
+        output = cont.logs(stdout=True, stderr=True)
+        # the startup script in the docker image starts postgres twice,
+        # so wait for the second start:
+        if output.count('database system is ready to accept connections') >= 2:
+            return True
+        time.sleep(0.1)
+    return False   # pragma: no cover
```

Comment:
The postgres docker image
1. starts postgres
2. creates the role according to the env variables
3. restarts postgres (dropping any connections)

My first attempt was to just wait until a connection to the db can be made, but in the tests, that would almost always end up making the connections just after 1., which would lead to dropped connections.
I did not want to use `wait_for_postgres` as this could mean waiting for 9 minutes, which I think is too long for a test to fail.
OTOH, this is only used in text code, so I'll move this to `tests`, which I hope makes it more acceptable, given how hacky it is.